packages/contracts/contracts/analytics-utilities/libs/QueryAssetHelpers.sol

```diff
@@ -8,175 +8,101 @@ import { ILendingPoolAddressesProvider } from "../../interfaces/ILendingPoolAddr
+        uint256 utilization;
+
+        // external rewards
+        address stakingContract;
+        uint256 totalStaked;    // the amount underlying staked by incentives controller (including rewards), same number of decimals as underlying token
+        uint256 pendingRewards; // pending rewards of incentives controller, same number of decimals as underlying token
+
+        // oracle
+        address oracle;
+        uint256 currentPriceETH; // current price of the underlying in eth (18 decimals)
     }
 
     function getAssetData(
         address asset,
         uint64 tranche,
-        address providerAddr,
-        address chainlinkOracleConverter
+        address providerAddr
     )
         internal
-        returns (AssetData memory assetData)
+        returns (AssetSummary memory assetSummary)
     {
         ILendingPool lendingPool = ILendingPool(
             ILendingPoolAddressesProvider(providerAddr).getLendingPool()
         );
+        IIncentivesController incentivesController = IIncentivesController(
+            ILendingPoolAddressesProvider(providerAddr).getIncentivesController()
+        );
 
         AssetMappings a = AssetMappings(ILendingPoolAddressesProvider(providerAddr).getAssetMappings());
 
-
         DataTypes.ReserveData memory reserve = lendingPool.getReserveData(asset, tranche);
-        assetData.tranche = tranche;
-        assetData.asset = asset;
-        (
-            assetData.ltv,
-            assetData.liquidationThreshold,
-            assetData.liquidationBonus,
-            assetData.decimals,
-            //borrowFactor (not used yet)
-        ) = a.getParams(asset, tranche);
-        assetData.canBeCollateral = reserve.configuration.getCollateralEnabled(asset, a);//assetData.liquidationThreshold != 0;
-        assetData.canBeBorrowed = reserve.configuration.getBorrowingEnabled(asset, a);
-        assetData.oracle = ILendingPoolAddressesProvider(providerAddr).getPriceOracle();
-        assetData.totalSupplied = convertAmountToUsd(assetData.oracle, assetData.asset, IAToken(reserve.aTokenAddress).totalSupply(), assetData.decimals, chainlinkOracleConverter);
-        assetData.totalBorrowed = convertAmountToUsd(assetData.oracle, assetData.asset, IAToken(reserve.variableDebtTokenAddress).totalSupply(), assetData.decimals, chainlinkOracleConverter);
-
-        assetData.totalReserves = convertAmountToUsd(assetData.oracle, assetData.asset, IERC20(asset).balanceOf(reserve.aTokenAddress), assetData.decimals, chainlinkOracleConverter);
-        assetData.totalReservesNative = IERC20(asset).balanceOf(reserve.aTokenAddress);
-
-        assetData.utilization = assetData.totalBorrowed == 0
+        assetSummary.reserveData = reserve;
+        assetSummary.tranche = tranche;
+        assetSummary.asset = asset;
+        assetSummary.decimals = IERC20(asset).decimals();
+
+        assetSummary.canBeCollateral = reserve.configuration.getCollateralEnabled(asset, a);
```

Comment:
I think you already know this but this is whether or not the reserve can be collateral, not whether or not the asset is allowed to be collateral

Reply:
yes, unfortunately the namign is a little bad. i will fix this to avoid future confusion